### DISCUSSION TITLE
Is it possible to use multiple material variants in one model?

### DISCUSSION THREAD

**Discussion by StefanOber** [google/model-viewer#3822] [Q&A]
Hi there,

the documentation only shows how to get the material variants of a model as an array of strings.
I was wondering if it is possible to access material variants on sub meshes. 
Let's say I have a car and one set of materials for the body and one for the seats.

Thanks!

**Comment by WadeMorrison** [comment]
Hey there Stefan!

This is possible to do, however from my expertise from doing it, you have to make sure they are separate meshes that have their own material(s).

**Reply by StefanOber** [reply]
Hi Wade,

thanks for the reply and great to hear it is possible! Could you give me a short example on how to access these variant groups? 
The docs only show how to get variants from the main model using modelViewer.availableVariants;. Or set a variant with modelViewer.variantName = "...". 
The variants we use are on sub meshes, each with its own material. 

Thanks!

**Reply by WadeMorrison** [reply]
Yeah of course! Here is a [early WIP](https://cashcredenza.s3.us-east-2.amazonaws.com/index.html) of what I made a long time ago, I've since been going in a different direction, so I haven't touched it in a while (the model is not optimized so loading times will be a little long). 

Basically it's this little JS snippet 
```
document.body.addEventListener('click', function (event) {
  if (event.target.className === 'item') {
    modelViewer.variantName = event.target.value;
  }
}, false);
```

and this HTML
```
<div id="items">
    <button class='item' value='Ballet Pink'>Ballet Pink</button>
    <button class='item' value="Bellini Peach">Bellini Peach</button>
</div>
```

In short, I have buttons that have a class name as "item" with a value of the variant name. If the person selects the button of Bellini Peach, that value will then equal the variant name in the GLB and assign that color instead. Hopefully that makes sense?

**Reply by elalish** [reply]
@StefanOber I think you may just misunderstand how glTF variants work. The material variants are defined on the sub meshes, but are collected through a top-level variant name. Our API only needs to select this top-level name, as then all the sub-mesh materials are automatically updated according to the extension data in the file, which may be a complex mapping.

**Reply by StefanOber** [reply]
@elalish Thanks for clarifying! So I'm bound to one set of variants. I was just hoping there might be the possibility of a sub-grouping of variants. Basically I would like to build a configurator.

**Reply by elalish** [reply]
To have mix and match permutations, I'd recommend you author with a separate material for each thing you want to change, then update their textures manually, as in https://modelviewer.dev/examples/scenegraph/#createTexturesExample

**Reply by StefanOber** [reply]
Okay thank you. I will give that a try.
